extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -138,11 +141,15 @@ export class BookTocManager implements IBookTocManager {
 			contents.forEach(async (content) => {
 				if ((await fs.stat(path.join(directory, content))).isFile) {
```

Comment:
You do this same path.join() call a couple times in this method, so you can probably cache it in a single variable.